### PR TITLE
Allow quick input to show info severity

### LINKED ISSUE #4591: "Forward a Port" command from command palette doesn't handle validation correctly
image

This should be an info, not an error, and you should be allowed to proceed past it. Generally, it should match the behavior in the Ports view:

image

### PR DESCRIPTION
**Pull request by alexr00**
Fixes microsoft/vscode-remote-release#4591

### CODE REVIEW

**Review by chrmarti — COMMENTED** [review]
Looks good. I have added two comments with the possibility of other code using this in mind. Please take a look.

File: src/vs/base/parts/quickinput/browser/quickInput.ts [hunk 1]
```diff
@@ -328,6 +352,13 @@ class QuickInput extends Disposable implements IQuickInput {
 		this.ui.ignoreFocusOut = this.ignoreFocusOut;
 		this.ui.setEnabled(this.enabled);
 		this.ui.setContextKey(this.contextKey);
+
+		const validationMessage = this.validationMessage || this.noValidationMessage;
+		if (this._lastValidationMessage !== validationMessage) {
+			this._lastValidationMessage = validationMessage;
+			dom.reset(this.ui.message, ...renderLabelWithIcons(escape(validationMessage)));
```
Comment (alexr00) [inline src/vs/base/parts/quickinput/browser/quickInput.ts:359]: When moving this from the subclasses to here, I saw that one sublcass ues `escape` and the other didn't. I opted to go with the `escape`, but I'm not 100% sure which is correct.

File: src/vs/base/parts/quickinput/browser/quickInput.ts [hunk 2]
```diff
@@ -1104,7 +1114,7 @@ class InputBox extends QuickInput implements IInputBox {
 		if (this._lastValidationMessage !== validationMessage) {
```
[the remark this answers is not in the document]
Comment (alexr00) [reply src/vs/base/parts/quickinput/browser/quickInput.ts]: Done!

File: src/vs/base/parts/quickinput/browser/quickInput.ts [hunk 3]
```diff
@@ -1056,6 +1058,14 @@ class InputBox extends QuickInput implements IInputBox {
 		this.update();
 	}
 
+	get severity() {
+		return this._severity;
+	}
+
+	set severity(severity: Severity) {
+		this._severity = severity;
```
[the remark this answers is not in the document]
Comment (alexr00) [reply src/vs/base/parts/quickinput/browser/quickInput.ts]: Also done!

**Comment by alexr00** [comment]
@chrmarti I made the changes. There is one thing I'm not sure about which I commented it.

File: src/vs/base/parts/quickinput/browser/quickInput.ts (hunk 1 above)
Comment (chrmarti) [reply src/vs/base/parts/quickinput/browser/quickInput.ts:360]: `escape` seems to make sense in both cases. 👍

File: src/vs/base/parts/quickinput/browser/quickInput.ts [hunk 4]
```diff
@@ -328,6 +352,13 @@ class QuickInput extends Disposable implements IQuickInput {
 		this.ui.ignoreFocusOut = this.ignoreFocusOut;
 		this.ui.setEnabled(this.enabled);
 		this.ui.setContextKey(this.contextKey);
+
+		const validationMessage = this.validationMessage || this.noValidationMessage;
+		if (this._lastValidationMessage !== validationMessage) {
+			this._lastValidationMessage = validationMessage;
+			dom.reset(this.ui.message, ...renderLabelWithIcons(escape(validationMessage)));
+			this.showMessageDecoration(this.severity);
```
Comment (chrmarti) [inline src/vs/base/parts/quickinput/browser/quickInput.ts:364]: I'll just move this to a separate `if` to ensure it updates when set after the validation message.

**Review by chrmarti — APPROVED** [review]
LGTM, thanks!

**Comment by alexr00** [comment]
Thanks for reviewing!

**Pull request merged**